lib/features/support/internal_hooks.rb

```diff
@@ -66,6 +60,18 @@
   # This must happen after any client hooks have run, so that they can set the server root
   Maze::DocumentServer.start unless Maze.config.document_server_root.nil?
 
+  # Determine public IP if enabled
+  if Maze.config.aws_public_ip
+    public_ip = Maze::AwsPublicIp.new
+    Maze.public_address = public_ip.address
+    $logger.info "Public address: #{Maze.public_address}"
+
+    unless Maze.config.document_server_root.nil?
+      Maze.public_document_server_address = public_ip.document_server_address
+      $logger.info "Public document server address: #{Maze.public_document_server_address}"
+    end
+  end
+
```

Comment:
This needed moving to after the hooks are run, so that any config we set is taken into account (`Maze.config.document_server_root` in this case).